crypto/zkp/schemes/cl/credential_issue_test.go

```diff
@@ -67,17 +67,28 @@ func TestCLIssue(t *testing.T) {
 	// (nonceUser, challenge, nymProofRandomData, nymProofData, UProofRandomData, UProofData, commitmentsOfAttrs,
 	// commitmentsOfAttrsProofs)
 
-	verified := orgCredentialIssuer.VerifyCredentialRequest(nymProofData, U, UProofData, commitmentsOfAttrsProofs)
+	verified := orgCredentialIssuer.VerifyCredentialRequest(credentialRequest)
 	assert.Equal(t, true, verified, "credential request sent to the credential issuer not correct")
 
 	nonceUser := userCredentialReceiver.GetNonce()
-	// TODO: implement credential struct when implementing update credential
-	A, e, v11, AProof := orgCredentialIssuer.IssueCredential(nonceUser)
+	credential, AProof := orgCredentialIssuer.IssueCredential(nonceUser)
 
-	userVerified, err := userCredentialReceiver.VerifyCredential(A, e, v11, AProof, nonceUser)
+	userVerified, err := userCredentialReceiver.VerifyCredential(credential, AProof, nonceUser)
 	if err != nil {
 		t.Errorf("error when verifying credential: %v", err)
 	}
 
 	assert.Equal(t, true, userVerified, "credential issuance failed")
+
+	nonceUser1 := userCredentialReceiver.GetNonce()
+	newKnownAttrs := []*big.Int{big.NewInt(17), big.NewInt(18), big.NewInt(19), big.NewInt(27)}
+	credential1, AProof1 := orgCredentialIssuer.UpdateCredential(nonceUser1, newKnownAttrs)
+	user.knownAttrs = newKnownAttrs
```

Comment:
Setting of user's `knownAttrs` should be "hidden away", or done somehow differently, IMO. Do you think it would make sense to add a `UpdateCredential` method on `UserCredentialReceiver` or `User`, just so that it's clear both sides need to update this value with a function call?

I tried commenting out L86 and the test failed, but to me it was not immediately obvious why. I see now that `UserCredentialReceiver` takes a pointer to `User`. It's a bit confusing, but I believe a pending TODO at L52 (concerning moving `UserCredentialReceiver` into `User`) can improve this. Still, perhaps adding `UpdateCredential` now isn't a bad idea.